docs/ansible/coding-conventions.md

```diff
@@ -0,0 +1,669 @@
+        msg: "fee foo faa"
+```
+
+ Why do this?
+
+A common order makes playbooks consistent and easier to read for your dear colleagues. Think of them when you write.
+
+## Order in task declaration
+
+A task should be declared in this order.
+
+* `name`
+* module and the arguments for the module
+* arguments for the task in alphabetic order
+* `tags` at the bottom
+
+```yaml
+---
+- name: unhold packages
+  ansible.builtin.shell: 'apt-mark -s unhold {{ item }} && apt-mark unhold {{ item }}'
+  args:
+    executable: /bin/bash
+  changed_when: apt_mark_unhold.rc == 0 and "already" not in apt_mark_unhold.stdout
+  failed_when: false
+  loop: '{{ packages_ubuntu_unhold }}'
+  register: apt_mark_unhold
+  tags:
+    - apt_unhold
+---
+```
+
+### Why do this?
+
+Reason is the same as "order of playbook". To make tasks more consistent and easier to read. Help your colleagues.
+
+## Air and tabulators
+
+Air, one of the **most important thing** for humans and **for code**! It must be an empty line before `vars`, `pre_tasks`, `roles` and `tasks`, and before each task in the tasks definition. Tabulator stops must be set to two, `2`, spaces.
+
+### Why do this?
+
+This creates a pretty and tidy code which is easy to read, both for dyslectic and non dyslectic people. There is no excuse not to do this.
+
+## Variable names
+
+Always use `snake_case` for variable names.
+
+✅ ***Good***
+
+```yaml
+---
+- name: set my variables
+  ansible.builtin.set_fact:
+    a_boolean: false
+    an_int: 101
+    a_string: bar
+```
+❌ ***Bad***
+
+```yaml
+---
+- name: set my variables
+  ansible.builtin.set_fact:
+    aBoolean: false
+    anint: 101
+    A_STRING: bar
+```
+ Why do this?
+
+Ansible already uses `snake_case` for variables in it's examples. Consistent naming of variables keeps the code tidy and gives better readability.
+
+## Content Organization
+
+Your usage of Ansible should fit your needs, however, not ours, so feel free to modify this approach and organize as you see fit.
+
+Always try to use Ansible’s “roles” to organize your playbook content  organization feature.
+
+## Directory Layout
+
+The top level of the directory would contain files and directories like so:
+
+```text
+production                # inventory file for production servers
+staging                   # inventory file for staging environment
+
+group_vars/
+  group1.yml              # here we assign variables to particular groups
+  group2.yml
+host_vars/
+  hostname1.yml           # here we assign variables to particular systems
+  hostname2.yml
+
+library/                  # if any custom modules, put them here (optional)
+  module_utils/           # if any custom module_utils to support modules, put them here (optional)
+  filter_plugins/         # if any custom filter plugins, put them here (optional)
+
+site.yml                  # master playbook
+webservers.yml            # playbook for webserver tier
+dbservers.yml             # playbook for dbserver tier
+
+  roles/
```

Comment:
for roles - separate section with file structure

mention that role may be generated with standard ansible galaxy command (it will create full file structure for the role). add command for quick copy/paste